common/components/collisions.py

```diff
@@ -0,0 +1,25 @@
+import enum
+from dataclasses import dataclass, field
+
+from common.engine import component
+from common.utils.vector2 import Vector2
+
+
+class CollisionDirection(enum.Enum):
+    UP = 0
+    DOWN = 1
+    LEFT = 2
+    RIGHT = 3
```

Comment:
`from enum import Enum, auto`

```suggestion
    UP = auto()
    DOWN = auto()
    LEFT = auto()
    RIGHT = auto()
```